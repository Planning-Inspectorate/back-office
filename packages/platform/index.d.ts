import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
