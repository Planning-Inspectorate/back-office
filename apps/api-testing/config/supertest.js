// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
