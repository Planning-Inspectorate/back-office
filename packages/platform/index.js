export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
