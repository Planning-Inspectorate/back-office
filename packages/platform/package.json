{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
