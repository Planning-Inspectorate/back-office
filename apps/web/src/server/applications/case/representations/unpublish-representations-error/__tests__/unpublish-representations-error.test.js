import { parseHtml } from '@pins/platform';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../testing/index.js';
import nock from 'nock';
import { fixtureCases } from '../../../../../../../testing/applications/fixtures/cases.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const baseUrl = '/applications-service/case/1/relevant-representations/unpublishing-error';

const nocks = () => {
	nock('http://test/').get('/applications/1').reply(200, fixtureCases[3]);
};

describe('unpublish-representations-error.controller', () => {
	describe('GET /applications-service/:caseId/relevant-representations/unpublishing-error', () => {
		beforeEach(installMockApi);
		afterEach(teardown);

		afterAll(() => {
			nock.cleanAll();
		});

		beforeEach(async () => {
			nocks();
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Sorry, there is a problem');
			expect(element.innerHTML).toContain(
				'None of your representations have been unpublished. Please try again later.'
			);
		});
	});
});
