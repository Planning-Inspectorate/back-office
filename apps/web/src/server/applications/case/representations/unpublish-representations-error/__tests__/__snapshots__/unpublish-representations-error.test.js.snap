// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unpublish-representations-error.controller GET /applications-service/:caseId/relevant-representations/unpublishing-error should render the page 1`] = `
"<main class="govuk-main-wrapper" id="main-content"><span class="govuk-caption-m"> Publishing queue</span>
    <h2 class="govuk-heading-l"> Sorry, there is a problem</h2>
    <p class="govuk-body">The publishing service is unavailable.</p>
    <p class="govuk-body">None of your representations have been unpublished. Please try again later.</p>
    <p     class="govuk-body">If you continue to have problems with this service, <a href="https://mhclg.service-now.com/sp"
        class="govuk-link">contact Littlefish</a>
        </p>
</main>"
`;
